pyanaconda/payload/rpmostreepayload.py

```diff
@@ -54,149 +53,57 @@ def source_type(self):
 
-        from pyanaconda.modules.payloads.payload.rpm_ostree.installation import \
-            PullRemoteAndDeleteTask
-        task = PullRemoteAndDeleteTask(data)
-        task.progress_changed_signal.connect(self._progress_cb)
-        task.run()
-
-        from pyanaconda.modules.payloads.payload.rpm_ostree.installation import DeployOSTreeTask
-        task = DeployOSTreeTask(data, conf.target.physical_root)
-        task.progress_changed_signal.connect(self._progress_cb)
-        task.run()
-
-        # Reload now that we've deployed, find the path to the new deployment
-        from pyanaconda.modules.payloads.payload.rpm_ostree.installation import SetSystemRootTask
-        task = SetSystemRootTask(conf.target.physical_root)
-        task.run()
-
-        from pyanaconda.modules.payloads.payload.rpm_ostree.installation import \
-            CopyBootloaderDataTask
-        task = CopyBootloaderDataTask(
-            sysroot=conf.target.system_root,
-            physroot=conf.target.physical_root
-        )
-        task.run()
-
-    def _prepare_mount_targets(self, data):
-        """ Prepare the ostree root """
-        from pyanaconda.modules.payloads.payload.rpm_ostree.installation import \
-            PrepareOSTreeMountTargetsTask
-        task = PrepareOSTreeMountTargetsTask(
-            sysroot=conf.target.system_root,
-            physroot=conf.target.physical_root,
-            source_config=data
-        )
-        bindmounts = task.run()
-        self._internal_mounts.extend(bindmounts)
+    def post_install(self):
+        """Run the post-installation tasks."""
+        task_paths = self.service_proxy.PostInstallWithTasks()
+        self._run_tasks(task_paths)
 
     def unsetup(self):
-        """Invalidate a previously setup payload."""
-        from pyanaconda.modules.payloads.payload.rpm_ostree.installation import \
-            TearDownOSTreeMountTargetsTask
-        task = TearDownOSTreeMountTargetsTask(
-            mount_points=self._internal_mounts
-        )
-        task.run()
+        """Tear down the sources and the payload."""
+        task_paths = self.service_proxy.TeardownWithTasks()
+        self._run_tasks(task_paths)
 
-        tear_down_sources(self.proxy)
+    def _progress_cb(self, step, message):
+        """Callback for task progress reporting."""
+        progressQ.send_message(message)
 
-    def post_install(self):
-        super().post_install()
-        data = self._get_source_configuration()
-
-        # Following up on the "remote delete" earlier, we removed the remote from
-        # /ostree/repo/config.  But we want it in /etc, so re-add it to /etc/ostree/remotes.d,
-        # using the sysroot path.
-        #
-        # However, we ignore the case where the remote already exists, which occurs when the
-        # content itself provides the remote config file.
-        #
-        # Note here we use the deployment as sysroot, because it's that version of /etc that we
-        # want.
-
```

Comment:
Shouldn't we keep this information somewhere? Maybe in post_install_with_tasks.
Not sure if it is still accurate and comprehensible though.

Reply:
The comment was moved to the `ChangeOSTreeRemoteTask` class.

Reply:
Thanks!